main.rb

```diff
@@ -0,0 +1,186 @@
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check == new_self.length
+  end
+
+  def my_any?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? true : false
+  end
+
+  def my_none?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? false : true
+  end
+
+  def my_count
+    return enum_for(:my_count) unless block_given?
+
+    self_length = length
+    i = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    length.times do
+      yield(new_self[i]) == false ? self_length -= 1 : self_length = self_length
+      i += 1
+    end
+    self_length
+  end
+
+  def my_map(my_proc = nil)
+    arr = []
+    i = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    if block_given?
+      new_self.length.times do
+        arr = arr << yield(new_self[i])
+        i += 1
+      end
+    elsif !my_proc.nil?
+      new_self.length.times do
+        arr = arr << my_proc.call(new_self[i])
+        i += 1
+      end
+    else
+      return enum_for(:my_map)
+    end
+    arr
+  end
+
+  def my_inject(sym = nil)
+    i = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    result = new_self[0]
+    if block_given?
+      while i < length - 1
+        result = yield(result, new_self[i + 1].to_i)
+        p result
+        i += 1
+      end
+    elsif !sym.nil?
+      sym = sym.to_s
+      while i < new_self.length - 1
+        self_elem = new_self[i + 1]
+        result = result.send(sym, self_elem.to_i)
+        i += 1
+      end
+    end
+    result
+  end
+end
```

Comment:
# my_inject
- [ ] `my_inject` raises a "LocalJumpError" when no block or argument is given

      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
        expected LocalJumpError but nothing was raised
##
- [ ] `my_inject` searches for the longest word in an array of strings
      Failure/Error: let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }

      NoMethodError:
        undefined method `length' for 0:Integer
##
- [ ] `my_inject` when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range

      Failure/Error: actual = range.my_inject { |prod, n| prod * n }

      NameError:
        undefined local variable or method `length' for 5..50:Range
##
- [ ] `my_inject` when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block

      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }

      NameError:
        undefined local variable or method `length' for 5..50:Range
##
- [ ] `my_inject` when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method

      Failure/Error: expect(array.my_inject(20, :*)).to eq array.inject(20, :*)

      ArgumentError:
        wrong number of arguments (given 2, expected 0..1)
##
- [ ] `my_inject` when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range

      Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)

      ArgumentError:
        wrong number of arguments (given 2, expected 0..1)